### ISSUE TITLE
Make recent submissions more accessible for screen readers

### ISSUE THREAD

**Issue by alexandrevryghem** [DSpace/dspace-angular#4268]
## Describe the bug
The recent submissions section on the home page currently uses a structure of multiple `<div>`s to render the recent submissions. This should be changed to an `<ul>`/`<li>` structure to be more accessible for screen readers. In addition, the "Load more ..." button should also be replaced with an `<a>` element to be more semantically accurate, since it always redirects you to the search page.

## Expected behavior
Screen readers should be able to detect that the recent submissions section displays a list of items, and it should be clear that the "Load more ..." button is actually not a button but a link to the search page.
